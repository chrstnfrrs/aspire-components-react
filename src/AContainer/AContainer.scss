.a-container {
  --AContainer-color: #000000;
  --AContainer-padding: 64px 128px;
  background: var(--A-background);
  color: var(--AContainer-color);
  padding: var(--AContainer-padding);
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
}
