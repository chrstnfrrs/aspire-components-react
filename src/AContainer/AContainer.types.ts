export interface AContainerProps {
  align?: string;
  className?: string;
  justify?: string;
}
