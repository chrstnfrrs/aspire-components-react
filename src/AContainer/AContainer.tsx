import React from 'react';
import clsx from 'clsx';
import { setJustify, setAlign } from 'aspire-components-functions';

import { AContainerProps } from './AContainer.types';
import './AContainer.scss';

const AContainer: React.FC<AContainerProps> = (props) => {
  const classList = [
    props.className,
    'a-container',
    setJustify(props.justify),
    setAlign(props.align),
  ];

  return <div className={clsx(classList)}>{props.children}</div>;
};

export default AContainer;
