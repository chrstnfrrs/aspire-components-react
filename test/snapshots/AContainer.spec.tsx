import AContainer from '../../src/AContainer/AContainer';

import { testComponent } from './helpers/component';
import { ALIGN_FACTORY, JUSTIFY_FACTORY } from './enums/factories';

describe('Given AContainer component is used', () => {
  const AContainerFactories = {
    factory: {
      align: ALIGN_FACTORY,
      justify: JUSTIFY_FACTORY,
    },
  };

  testComponent(AContainer, 'AContainer', AContainerFactories);
});
