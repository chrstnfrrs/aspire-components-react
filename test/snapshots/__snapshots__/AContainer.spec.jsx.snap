// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given AContainer component is used When a developer uses AContainer with a className Then the the class should be present 1`] = `
<div
  className="asdf a-container"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses AContainer with no props Then the only class should be a-container  1`] = `
<div
  className="a-container"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the align prop When a developer passes in an invalid value Then no align class should be used 1`] = `
<div
  className="a-container"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the align prop When a developer passes in baseline Then the class align-baseline should be used 1`] = `
<div
  className="a-container align-baseline"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the align prop When a developer passes in center Then the class align-center should be used 1`] = `
<div
  className="a-container align-center"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the align prop When a developer passes in end Then the class align-end should be used 1`] = `
<div
  className="a-container align-end"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the align prop When a developer passes in start Then the class align-start should be used 1`] = `
<div
  className="a-container align-start"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the align prop When a developer passes in stretch Then the class align-stretch should be used 1`] = `
<div
  className="a-container align-stretch"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the align prop When a developer passes in undefined Then no align class should be used 1`] = `
<div
  className="a-container"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in an invalid value Then no justify class should be used 1`] = `
<div
  className="a-container"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in around Then the class justify-around should be used 1`] = `
<div
  className="a-container justify-around"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in between Then the class justify-between should be used 1`] = `
<div
  className="a-container justify-between"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in center Then the class justify-center should be used 1`] = `
<div
  className="a-container justify-center"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in end Then the class justify-end should be used 1`] = `
<div
  className="a-container justify-end"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in evenly Then the class justify-evenly should be used 1`] = `
<div
  className="a-container justify-evenly"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in start Then the class justify-start should be used 1`] = `
<div
  className="a-container justify-start"
>
  AContainer
</div>
`;

exports[`Given AContainer component is used When a developer uses a factory prop type When a developer uses the justify prop When a developer passes in undefined Then no justify class should be used 1`] = `
<div
  className="a-container"
>
  AContainer
</div>
`;
